_extensions/redirects.py

```diff
@@ -0,0 +1,54 @@
+# Adapted from https://github.com/sphinx-contrib/redirects
+
+import os
+import re
+
+from sphinx.builders import html as builders
+from sphinx.util import logging as logging
+
+TEMPLATE = '<html><head><meta http-equiv="refresh" content="0; url=%s"/></head></html>'
+
+logger = logging.getLogger(__name__)
+
+def generate_redirects(app):
+    path = os.path.join(app.srcdir, app.config.redirects_file)
+    if not os.path.exists(path):
+        logger.info("Could not find redirects file at '%s'" % path)
+        return
+
+    source_suffix = next(iter(app.config.source_suffix))
+
+    if not type(app.builder) == builders.StandaloneHTMLBuilder:
+        logger.info("Redirects are only supported by the 'html' builder. Skipping...")
+        return
+
+    with open(path) as redirects:
+        pattern = re.compile(r"^([\w./ ]+)(#.*)?$")
+        for line in redirects.readlines():
+            match_result = pattern.search(line)
+            if not match_result:
+                continue
+
+            redirect = match_result.group(1).rstrip()
+            if redirect.count(' ') != 1:
+                logger.error("Ignoring malformed redirection: %s" % redirect)
+                continue
+            from_path, to_path = redirect.split()
+            logger.debug("Redirecting '%s' to '%s'" % (from_path, to_path))
+
+            from_path = from_path.replace(source_suffix, '.html')
+            to_path_prefix = '..%s' % os.path.sep * (len(from_path.split(os.path.sep)) - 1)
```

Comment:
this assume we support both `\` and `/` as separator depending of the plateform? I think we should force to use `/` in the `from_path`.
Have you tried using [pathlib](https://docs.python.org/3/library/pathlib.html)? I think it can clarify the code (and can handles all the `os` operations).

Reply:
I considered pathlib but it was added in Python 3.4. Do we want to have this restriction all of sudden when we just updated our requirements to just "Python 3" ?

Ok for enforcing `/` in the from_path.

Reply:
In fact I removed this line completely and replaced it with `absolute_to_path = os.path.join(app.builder.outdir, to_path)` so that both paths start from the root, which makes much more sense.

Reply:
> I considered pathlib but it was added in Python 3.4. Do we want to have this restriction all of sudden when we just updated our requirements to just "Python 3" ?

We [never supported python below 3.5](https://www.odoo.com/groups/community-59/community-32859400) afaik